test/utils/nrosched/nrosched.go

```diff
@@ -52,18 +56,32 @@ func CheckPODWasScheduledWith(k8sCli *kubernetes.Clientset, podNamespace, podNam
 		klog.ErrorS(err, "cannot get events for pod %s/%s", podNamespace, podName)
 		return false, err
 	}
+
 	for _, item := range events.Items {
 		klog.Infof("checking event: [%s: %s]", item.ReportingController, item.Reason)
-		// TODO: is this check robust enough?
-		if item.Reason == ReasonScheduled && item.ReportingController == schedulerName {
-			klog.Infof("-> found relevant scheduling event for pod %s/%s", podNamespace, podName)
+		if evCheck(item) {
+			klog.Infof("-> found relevant scheduling event for pod %s/%s: %v", podNamespace, podName, item)
 			return true, nil
 		}
 	}
 	klog.Warningf("Failed to find relevant scheduling event for pod %s/%s", podNamespace, podName)
 	return false, nil
 }
 
+func CheckPODSchedulingFailed(k8sCli *kubernetes.Clientset, podNamespace, podName, schedulerName string) (bool, error) {
```

Comment:
This function doesn't cover for pod which stuck ```Pending``` state which is also included under the "scheduling failed" category.

Reply:
Good point! that's intentional. Pod in `Pending` state are'nt technically failed from scheudling POV. So I think we should handle them separately with a new helper.